.github/workflows/spelling.yml

```diff
@@ -92,6 +92,8 @@ jobs:
     steps:
     - name: Checkout code
       uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request && format('refs/pull/{0}/merge', github.event.pull_request.number) || github.event.ref }}
```

Comment:
check-spelling has a feature built-in to handle this stuff, but we opted to use the official `actions/checkout` action which means we have to explicitly tell checkout that we want to get the merge for PRs (it defaults to the head of the PR base branch for `pull_request_target`).

Reply:
@jsoref 🖖 action/spelling is still failing. Any idea?

Reply:
This is because the version we're running is the one in `ceph:devel` not the one we're pull-requesting. 

You can see this working in https://github.com/check-spelling/ceph-nvmeof/pull/1#event-10307489016 (for ❌ ) and the latest commit of https://github.com/check-spelling/ceph-nvmeof/pull/1 for (✅)

Reply:
@jsoref 🖖 Thank you for explaining. Sounds like the `with:` clause should be removed for checkout action then. 
[Looks good in my fork](https://github.com/baum/ceph-nvmeof/actions/runs/6123431274).

Reply:
No, removing the `with` won't help. You need it in order to check out the PR head merge.

But you can't test changes to this workflow in a PR you have to test using the branch w/ the changes as the base branch.

Reply:
In which scenario omitting `with` would not checkout the correct version? How do you suggest to move forward?

Reply:
Merge this PR w/o your additional commit, it'll do the right thing once merged.

Without the with PRs will never check the content of the PR. w/ the with, PRs _after_ the with is merged will check the content of the PR.

Reply:
Tried removing the `with`. Why the PR check fails? I am confused 🤷‍♂️.  Can we get the CI checks green before merging?

Reply:
Nope. Because of the way `on: pull_request_target` works the workflow file that's running is the one in the destination branch.

Reply:
I am still confused about  🤷‍♂️  about an idea of merging a change that does not pass. That makes it tricky to validate. I usually do not use `with ref` on checkouts,  that results in expected codebase and tightly coupled corresponding workflow tests in all scenarios (pr, push to branch whatever), but I might be missing something.

Reply:
That's part of why I have a `checkout:` feature in check-spelling, it's too hard to reason over the `actions/checkout` step with `pull_request_target`.

Most of the time the workflow doesn't change. For when it does/needs to change (as here), the way to test it is to:
1. create a branch (in this repository or another) with the changed workflow
2. create a branch from that branch (in that repo, or another) and create a test PR into that first branch
3. check to make sure the test PR shows the expected results
4. add a typo to the second branch
5. check to make sure the test PR shows the expected results

I have a bit of time for the next ~ hour to talk w/ you by slack...

Reply:
@jsoref 🖖 I want to express my gratitude for your dedicated efforts.

I'm not entirely certain about the best approach, but let's explore the possibility of accomplishing the following objectives:

1. The spelling workflow test should consistently assess the current state (revision) of the codebase.
2. The spelling workflow test should utilize the existing spelling configuration located in the action/spelling directory within this revision.
3. We should ensure that the spelling test uses the fixed version of the check-spelling/check-spelling action.

While we seem to be on track with point 3, points 1 and 2 are still under discussion. In the meantime, I'll submit a PR to temporarily disable the spelling workflow until we can fully address these objectives. It appears that the spelling workflow requires further attention, and having consistently failing checks is causing some unease.